Gifski/AppDelegate.swift

```diff
@@ -297,16 +280,32 @@ final class AppDelegate: NSObject, NSApplicationDelegate {
 		circularProgress.progress = 0
 		circularProgress.animated = true
 
-		progress = Progress(totalUnitCount: 1)
-		progress?.becomeCurrent(withPendingUnitCount: 1)
-		gifski = Gifski.convertFile(
-			inputUrl,
-			outputUrl: outputUrl,
-			quality: defaults["outputQuality"] as! Double,
+		let progress = Progress(totalUnitCount: 1)
+		progress.becomeCurrent(withPendingUnitCount: 1)
+		Gifski.convert(
+			fileAt: inputUrl,
+			outputTo: outputUrl,
+			withQuality: defaults["outputQuality"] as! Double,
 			dimensions: choosenDimensions,
 			frameRate: choosenFrameRate
-		)
-		progress?.resignCurrent()
+		) { [weak self] result in
+			DispatchQueue.main.async {
+				switch result {
+				case .success:
+					self?.circularProgress.percentLabelLayer.string = "✔"
+					self?.circularProgress.fadeOut(delay: 1) {
+						self?.isRunning = false
+					}
+				case .error(let error):
+					fatalError(error.localizedDescription)
+				}
```

Comment:
What do you think about something like the follow:

```swift
extension Result {
  func unwrap() -> T {
    switch self {
    case .success(let value):
      return value
    case .error(let error):
      fatalError(error.localizedDescription)
    }
  }
}
```

I would simplify this code.

Reply:
Yeah, look better.
Maybe forceUnwrap() to indicate that it will fatall on error?

This error type has a lot of nice helpers: https://github.com/antitypical/Result

Eventually these errors should probably not fatal. Just show an error message, log and allow the user to try again.

Reply:
> Maybe forceUnwrap() to indicate that it will fatall on error?

👍 

> This error type has a lot of nice helpers: antitypical/Result

It's good, but too much for our simple needs, I think.